tests/telex.rs

```diff
@@ -0,0 +1,39 @@
+use std::fmt::Write;
+
+use serde::Serialize;
+
+fn snapshot_transform(input: &str) -> String {
+    let mut result = String::new();
+    for line in input.lines() {
+        let words = line.trim().split_whitespace();
+
+        let mut transformed_line = String::new();
+        for word in words {
+            let transformed_words = vi::telex::transform_buffer(word.chars());
+            write!(&mut transformed_line, "{} ", transformed_words).unwrap();
+        }
+        write!(&mut result, "{}\n", transformed_line.trim()).unwrap();
+    }
+    result
+}
+
+#[derive(Serialize)]
+struct Metadata<'a> {
+    input_file: &'a str
+}
+
+macro_rules! gen_test {
```

Comment:
It'd be great if we can separate `gen_test` into a generic macro that can be used for both `vni` & `telex` tests. Would it be possible to do that? 🤔

Reply:
definitely

Reply:
Great! Do you want to introduce that change now? Otherwise I can introduce it once I add tests for vni method.

Reply:
nvm haha 😄